File: routes/dietRoutes.js
import express from "express";
import { generateDiet, swapMeal } from "../controllers/dietController.js";
import { protect } from "../middleware/authMiddleware.js";
import { validateDietGeneration } from "../middleware/validateRequest.js";

const router = express.Router();

router.post("/generate", protect, validateDietGeneration, generateDiet);
router.post("/swap", protect, swapMeal);

export default router;

File: routes/diet_routes.py
from fastapi import APIRouter, Depends
from controllers.diet_controller import generate_diet, swap_meal, GenerateDietRequest, SwapMealRequest
from middleware.auth_middleware import get_current_user

router = APIRouter(prefix="/api/diet", tags=["Diet"])


@router.post("/generate")
async def generate_diet_route(
    data: GenerateDietRequest,
    user_id: str = Depends(get_current_user)
):
    return await generate_diet(data, user_id)


@router.post("/swap")
async def swap_meal_route(
    data: SwapMealRequest,
    user_id: str = Depends(get_current_user)
):
    return await swap_meal(data, user_id)
